pandas/core/computation/eval.py

```diff
@@ -232,8 +237,13 @@ def eval(expr, parser='pandas', engine=None, truediv=True,
     pandas.DataFrame.query
     pandas.DataFrame.eval
     """
-    inplace = validate_bool_kwarg(inplace, 'inplace')
-    first_expr = True
+
+    inplace = validate_bool_kwarg(inplace, "inplace")
+    modifiable = hasattr(target, "__setitem__")
+
+    if inplace and not modifiable:
+        raise ValueError("Cannot modify the provided target inplace")
```

Comment:
Just using both as keyword arguments: pd.eval(expr, target=df, inplace=True)

that only makes sense when expr contains an assignment, and that you check with `parsed_expr.assigner`